### PR TITLE
feat!: upgrade binaryen to 0.4.0 & update test invocation to work with it

### PR DESCRIPTION
**Pull request by phated**
This bumps Binaryen.ml to 0.4.0 and fixes the test invocation so the tests will run on Windows.

I've marked this as a breaking change since technically we lose support for things like `x86_32` operating systems by vendoring binaryen.

### CODE REVIEW

**Comment by phated** [comment]
Something seems to be very wrong with the tests.

**Comment by phated** [comment]
It seems that `esy b dune runtest` broke the tests on Mac and Linux. I removed it to see if this works in CI (it works locally again). However, that is likely needed for Windows, so I need to investigate further.

File: compiler/esy.json [hunk 1]
```diff
@@ -15,12 +15,12 @@
   "scripts": {
     "copy-compiler": "cp #{$GRAINC_BIN_PATH} #{$CLI_BIN_DIR}",
     "clean": "rm -rf #{self.root}/_esy",
-    "test": "dune runtest --force",
+    "test": "dune runtest",
```
Comment (ospencer) [inline compiler/esy.json:18]: Any way we can keep `--force`? It's useful for when you change the runtime and want to re-run the tests.

Comment (phated) [reply compiler/esy.json]: I didn't know what this did 🙈 Re-added

**Review by ospencer — APPROVED** [review]
Nice!

**Pull request merged**